plugins/package-managers/bazel/src/main/kotlin/Bazel.kt

```diff
@@ -0,0 +1,309 @@
+
+import java.io.File
+
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.json.Json
+
+import org.apache.logging.log4j.kotlin.logger
+
+import org.ossreviewtoolkit.analyzer.AbstractPackageManagerFactory
+import org.ossreviewtoolkit.analyzer.PackageManager
+import org.ossreviewtoolkit.clients.bazelmoduleregistry.BazelModuleRegistryClient
+import org.ossreviewtoolkit.clients.bazelmoduleregistry.ModuleMetadata
+import org.ossreviewtoolkit.clients.bazelmoduleregistry.ModuleSourceInfo
+import org.ossreviewtoolkit.downloader.VersionControlSystem
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageLinkage
+import org.ossreviewtoolkit.model.PackageReference
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+import org.ossreviewtoolkit.utils.common.CommandLineTool
+
+import org.semver4j.RangesList
+import org.semver4j.RangesListFactory
+
+import retrofit2.HttpException
+
+const val BZLMOD_LOCK_FILE = "MODULE.bazel.lock"
+
+private val json = Json {
+    ignoreUnknownKeys = true
+}
+
+class Bazel(
+    name: String,
+    analysisRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : PackageManager(name, analysisRoot, analyzerConfig, repoConfig), CommandLineTool {
+    class Factory : AbstractPackageManagerFactory<Bazel>("Bazel") {
+        override val globsForDefinitionFiles = listOf("MODULE", "MODULE.bazel")
+
+        override fun create(
+            analysisRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) = Bazel(type, analysisRoot, analyzerConfig, repoConfig)
+    }
+
+    override fun command(workingDir: File?) = "bazel"
+
+    override fun getVersionArguments() = "version"
+
+    override fun transformVersion(output: String) = transformBazelVersion(output)
+
+    // Bazel 6.0 already supports bzlmod but it is not enabled by default.
+    // Supporting it would require adding the flag "--enable_bzlmod=true" at the correct position of all bazel
+    // invocations.
+    override fun getVersionRequirement(): RangesList = RangesListFactory.create(">=7.0")
+
+    override fun resolveDependencies(definitionFile: File, labels: Map<String, String>): List<ProjectAnalyzerResult> {
+        val projectDir = definitionFile.parentFile
+        val lockFile = projectDir.resolve(BZLMOD_LOCK_FILE)
+        requireLockfile(projectDir) { lockFile.isFile }
+
+        val projectVcs = processProjectVcs(projectDir)
+        val moduleMetadata = Parser(definitionFile.readText()).parse()
+        val depDirectives = moduleMetadata.dependencies.associateBy { "${it.name}@${it.version}" }
```

Comment:
Is it possible that a single module (name) can occur in the dependency try with multiple times with different versions ?

Reply:
The code could be simplified to associate the entries just by name. 
This would remove the (for me) open question, what happens in case of an empty version string.
E.g. could the current code swallow some entries by associating different meteadata with the same key in that case?

Reply:
The reason I'm reshuffling `moduleMetadata.dependencies` into this map is because Bazel uses the format `name@version` in the list of dependencies in the output of `bazel mod graph` and in the lock file. The map is used to split the direct dependencies into `main` and `dev` scope.